---
title: "Class 10: Structural Bioinformatics pt1"
author: "Kevin  (A16482696)"
format: pdf
---

## What is the PDB database

The main repository of biomolecular structure info is the PDB <www.rcsd.org>.

Let's see what this data base contains:

```{r}
stats <- read.csv("pdb_stats.csv", row.names = 1)
stats
```


> Q1: What percentage of structures in the PDB are solved by X-Ray and Electron Microscopy.

84% are solved by xray and EM solves 8.7%

```{r}
as.numeric(stats$X.ray)
```

We got to get rid of the commas. Can you find a function to get rid of the commas?

```{r}
x <- stats$X.ray
sum(as.numeric(gsub(",", "", x)))
```

I am going to turn this into a function and then use `apply()` to work on the entire table of data

```{r}
sumcomma <- function(x) { 
  sum(as.numeric(gsub(",", "", x)))
}

sumcomma(stats$X.ray)
```

```{r}
sumcomma(stats$Total)
```

```{r}
n.total <- sumcomma(stats$Total)
n.total
```

```{r}
sumcomma(stats$EM)
```

```{r}
apply(stats,2, sumcomma)
```

```{r}
apply(stats,2, sumcomma) / sumcomma(stats$Total)
```


> Q2: What proportion of structures in the PDB are protein?

In UniProt there are 248,805,733 entries which compared to PDB protein entries (186898) means there are only 7% of known sequences with a known structure 


> Q3: Type HIV in the PDB website search box on the home page and determine how many HIV-1 protease structures are in the current PDB?

5

>Q4: Water molecules normally have 3 atoms. Why do we see just one atom per water molecule in this structure?

the hydrogen is too small to be shown

>Q5: There is a critical “conserved” water molecule in the binding site. Can you identify this water molecule? What residue number does this water molecule have

D25

>Q6: Generate and save a figure clearly showing the two distinct chains of HIV-protease along with the ligand. You might also consider showing the catalytic residues ASP 25 in each chain and the critical water (we recommend “Ball & Stick” for these side-chains). Add this figure to your Quarto document.

## Visualizing the HIV-1 protease structure

Mol* (mol-star) viewer is now everywhere. The homepage is here:
https://molstar.org/viewer/.

I want to insert my image from Mol* here.

![My first molecular image](1HSG.png)

## Working iwht ht ebio3d package 

```{r}
library(bio3d)

```

```{r}
pdb <- read.pdb("1hsg")
pdb
```
```{r}
head(pdb$atom)
```

```{r}
pdbseq(pdb)[25]
```


## Predicting functional motions of a single structure

We can do bioinformatics prediction of functional motions (i.e. flexibility/dynamics)

```{r}
pdb <- read.pdb("6s36")
pdb
```

```{r}
m <- nma(pdb)
plot(m)
```

>Q7: How many amino acid residues are there in this pdb object? 

214

>Q8: Name one of the two non-protein residues? 

CL (3)

>Q9: How many protein chains are in this structure? 

1

```{r}
mktrj(m, file="adk_m7.pdb")
```


![](ADK_M7.PDB.png)

# Comparative Analysis of Protien Structures

Here we will work throguh a complete pipline of analysis for a given protein family starting from one single database ID for a member of this family all the way to a deeper understanding of structure, dynamics and functions.

```{r}
library(bio3d)

```


```{r}
id <- "1ake_A"

aa <- get.seq(id)
aa
```



```{r}
# Download releated PDB files
hits <- NULL
hits$pdb.id <- c('1AKE_A','6S36_A','6RZE_A','3HPR_A','1E4V_A','5EJE_A','1E4Y_A','3X2S_A','6HAP_A','6HAM_A','4K46_A','3GMT_A','4PZL_A')
files <- get.pdb(hits$pdb.id, path="pdbs", split=TRUE, gzip=TRUE)
```

```{r}
# Align releated PDBs
pdbs <- pdbaln(files, fit = TRUE, exefile="msa")
```

print out the alignment 
```{r}
pdbs
```


```{r}
# Perform PCA
pc.xray <- pca(pdbs)
plot(pc.xray)
```
```{r}
# Visualize first principal component
pc1 <- mktrj(pc.xray, pc=1, file="pc_1.pdb")

```









